### PR TITLE
adding BOOST_NO_EXCEPTIONS around file_storage add files

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

**Comment by arvidn** [comment]
are you actually building libtorrent with `-fno-exceptions`? does it work?

**Comment by aldenml** [comment]
not really, I just use it as a safeguard for not use the API that throws while using the SWIG interface. Do you want to remove these checks?

**Comment by arvidn** [comment]
well, I've been tempted to either remove them or fix support for building without exceptions (where "support" means terminating)

**Comment by aldenml** [comment]
iirc, the problem is not only in `libtorrent` but in `boost` also, meaning that even using `BOOST_NO_EXCEPTIONS`, boost requires exceptions support due to the use of some related keywords, but I haven't tried it recently.

File: include/libtorrent/file_storage.hpp [hunk 1]
```diff
@@ -306,6 +307,7 @@ namespace aux {
 			, file_flags_t file_flags = {}
 			, std::time_t mtime = 0, string_view symlink_path = string_view()
 			, char const* root_hash = nullptr);
+#endif
```
Comment (arvidn) [inline include/libtorrent/file_storage.hpp]: you would have to surround the definitions of these functions with this preprocessor condition too, right?

Comment (arvidn) [reply include/libtorrent/file_storage.hpp]: otherwise I imagine you would get a compile error

Comment (aldenml) [reply include/libtorrent/file_storage.hpp]: that's right!

Comment (aldenml) [reply include/libtorrent/file_storage.hpp]: done

**Pull request merged**